paddle/fluid/operators/controlflow/fetch_op.cc

```diff
@@ -36,53 +62,43 @@ class FetchOp : public framework::OperatorBase {
     PADDLE_ENFORCE(fetch_var != nullptr,
                    "Cannot find fetch variable in scope, fetch_var_name is %s",
                    fetch_var_name);
-
     auto out_name = this->Output("Out");
     auto *out_var = scope.FindVar(out_name);
     PADDLE_ENFORCE(out_var != nullptr,
                    "Cannot find out_var in scope, out_var_name is %s",
                    out_name);
-
     auto col = static_cast<size_t>(Attr<int>("col"));
-
-    auto *fetch_list = out_var->GetMutable<framework::FeedFetchList>();
-    auto &src_item = fetch_var->Get<framework::FeedFetchType>();
-
-    if (col >= fetch_list->size()) {
-      fetch_list->resize(col + 1);
-    }
-    auto &dst_item = fetch_list->at(col);
-
-    // FIXME(yuyang18): Should we assume the fetch operator always generate
-    // CPU outputs?
-    if (src_item.IsInitialized() && src_item.numel() > 0) {
-#ifdef PADDLE_WITH_MKLDNN
-      // Conversion from MKL-DNN to Paddle
-      if (src_item.layout() == framework::DataLayout::kMKLDNN) {
-        framework::Tensor out;
-        // Convert to desired Paddle layout, apart from grads of filter
-        // as params are not a subject to paddle's data_format
-        framework::innerTransDataLayoutFromMKLDNN(
-            src_item.layout(),
-            fetch_var_name == framework::GradVarName("Filter")
-                ? framework::DataLayout::kNCHW
-                : paddle::platform::get_cur_paddle_data_layout(),
-            src_item, &out, platform::CPUPlace());
-        TensorCopySync(out, platform::CPUPlace(), &dst_item);
-      } else {
-        TensorCopySync(src_item, platform::CPUPlace(), &dst_item);
+    if (out_var->IsType<framework::FeedFetchList>()) {
+      auto *fetch_list = out_var->GetMutable<framework::FeedFetchList>();
+      if (col >= fetch_list->size()) {
+        fetch_list->resize(col + 1);
       }
-#else
-      TensorCopySync(src_item, platform::CPUPlace(), &dst_item);
-#endif
+      auto *dst_item = &(fetch_list->at(col));
```

Comment:
Using auto when the type is obvious, otherwise you should specify the type in C++

For example, `auto &src_item = fetch_var->Get<framework::FeedFetchType>()` is very clear that `src_item` is a reference of `framework::FeedFetchType`, so that's good.

But not `auto *dst_item = &(fetch_list->at(col));`